From 63c1080052a48b6efcdf0e533c1745d29d415429 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Mon, 26 May 2014 16:40:24 +0800
Subject: [PATCH 320/602] ENGR00306120 - [Android_MX6SL_EVK] System: System
 reboot occurred after

longtime stress test.

Seems the watchdog on evk6sl may has issue after long time stress test.
Pre-timeout is trigged less than 15 seconds.
Before fix the issue, disable watchdogd in userspace to walk around.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 arm2_6dq/init.rc       | 4 ++++
 arm2_6sl/init.rc       | 4 ++++
 hdmidongle_6dq/init.rc | 4 ++++
 imx6/etc/init.rc       | 3 ---
 sabreauto_6q/init.rc   | 4 ++++
 sabresd_6dq/init.rc    | 4 ++++
 6 files changed, 20 insertions(+), 3 deletions(-)

diff --git a/arm2_6dq/init.rc b/arm2_6dq/init.rc
index bfb1f0f..f40a5d8 100755
--- a/arm2_6dq/init.rc
+++ b/arm2_6dq/init.rc
@@ -74,6 +74,10 @@ service wpa_supplicant /system/bin/wpa_supplicant \
     disabled
     oneshot
 
+# Set watchdog timer to 30 seconds and pet it every 10 seconds to get a 20 second margin
+    service watchdogd /sbin/watchdogd 10 20
+    class core
+
 on fs
 # mount ext4 partitions
     mount ext4 /dev/block/mmcblk0p5 /system
diff --git a/arm2_6sl/init.rc b/arm2_6sl/init.rc
index c651075..bccf631 100755
--- a/arm2_6sl/init.rc
+++ b/arm2_6sl/init.rc
@@ -67,6 +67,10 @@ service wpa_supplicant /system/bin/wpa_supplicant \
     disabled
     oneshot
 
+# Set watchdog timer to 30 seconds and pet it every 10 seconds to get a 20 second margin
+    service watchdogd /sbin/watchdogd 10 20
+    class core
+
 on fs
 # mount ext4 partitions
     mount ext4 /dev/block/mmcblk0p5 /system
diff --git a/hdmidongle_6dq/init.rc b/hdmidongle_6dq/init.rc
index 69784bc..1ce0d2e 100755
--- a/hdmidongle_6dq/init.rc
+++ b/hdmidongle_6dq/init.rc
@@ -128,6 +128,10 @@ service iprenew_bt-pan /system/bin/dhcpcd -n
     disabled
     oneshot
 
+# Set watchdog timer to 30 seconds and pet it every 10 seconds to get a 20 second margin
+    service watchdogd /sbin/watchdogd 10 20
+    class core
+
 on fs
 # mount ext4 partitions
     mount_all /fstab.freescale
diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 055b0bd..5159729 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -725,6 +725,3 @@ service wifi_mac /system/bin/sh /system/etc/check_wifi_mac.sh
     service sdcard /system/bin/sdcard -l -t4 /data/media /mnt/shell/emulated 1023 1023
     class late_start
 
-# Set watchdog timer to 30 seconds and pet it every 10 seconds to get a 20 second margin
-service watchdogd /sbin/watchdogd 10 20
-    class core
diff --git a/sabreauto_6q/init.rc b/sabreauto_6q/init.rc
index 91ae165..ab0e260 100644
--- a/sabreauto_6q/init.rc
+++ b/sabreauto_6q/init.rc
@@ -154,6 +154,10 @@ service ing /system/bin/ingsvcd -c /system/etc/gps/Orion.ini
     group gps
     oneshot
 
+# Set watchdog timer to 30 seconds and pet it every 10 seconds to get a 20 second margin
+service watchdogd /sbin/watchdogd 10 20
+    class core
+
 on fs
 # mount ext4 partitions
     mount_all /fstab.freescale
diff --git a/sabresd_6dq/init.rc b/sabresd_6dq/init.rc
index 79d6ccd..bd56190 100755
--- a/sabresd_6dq/init.rc
+++ b/sabresd_6dq/init.rc
@@ -108,6 +108,10 @@ on boot
 # mount the debugfs
     mount debugfs none /sys/kernel/debug/
 
+# Set watchdog timer to 30 seconds and pet it every 10 seconds to get a 20 second margin
+    service watchdogd /sbin/watchdogd 10 20
+    class core
+
 on charger
     # Enable Power modes and set the CPU Freq Sampling rates
     write /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor "powersave"
-- 
1.8.0

